contracts/social-tips/src/contract.rs

```diff
@@ -0,0 +1,1346 @@
+const ATTRIBUTE_NEW_ADMIN: &str = "new_admin";
+const ACTION_INSTANTIATE: &str = "instantiate";
+const ACTION_SEND_TIPS: &str = "send_tips";
+const ACTION_UPDATE_ADMIN: &str = "update_admin";
+const ACTION_CLAIM_PENDING_TIPS: &str = "claim_pending_tips";
+const ACTION_UPDATE_MAX_PENDING_TIPS: &str = "update_max_pending_tips";
+const ACTION_UPDATE_MAX_SENT_PENDING_TIPS: &str = "update_max_sent_pending_tips";
+const ACTION_REMOVE_PENDING_TIP: &str = "remove_pending_tip";
+
+#[cfg_attr(not(feature = "library"), entry_point)]
+pub fn instantiate(
+    deps: DepsMut<DesmosQuery>,
+    _env: Env,
+    info: MessageInfo,
+    msg: InstantiateMsg,
+) -> Result<Response<DesmosMsg>, ContractError> {
+    msg.validate()?;
+    set_contract_version(deps.storage, CONTRACT_NAME, CONTRACT_VERSION)?;
+
+    let admin = if let Some(address) = msg.admin {
+        deps.api.addr_validate(&address)?
+    } else {
+        info.sender
+    };
+
+    CONFIG.save(
+        deps.storage,
+        &Config {
+            admin,
+            max_pending_tips: msg.max_pending_tips,
+            max_sent_pending_tips: msg.max_sent_pending_tips,
+        },
+    )?;
+
+    Ok(Response::new().add_attribute(ATTRIBUTE_ACTION, ACTION_INSTANTIATE))
+}
+
+#[cfg_attr(not(feature = "library"), entry_point)]
+pub fn execute(
+    deps: DepsMut<DesmosQuery>,
+    env: Env,
+    info: MessageInfo,
+    msg: ExecuteMsg,
+) -> Result<Response<DesmosMsg>, ContractError> {
+    msg.validate()?;
+
+    match msg {
+        ExecuteMsg::SendTip {
+            application,
+            handle,
+        } => send_tip(deps, env, info, application, handle),
+        ExecuteMsg::ClaimTips {} => claim_tips(deps, info),
+        ExecuteMsg::UpdateAdmin { new_admin } => update_admin(deps, new_admin),
+        ExecuteMsg::UpdateMaxPendingTips { value } => update_max_pending_tips(deps, info, value),
+        ExecuteMsg::UpdateMaxSentPendingTips { value } => {
+            update_max_sent_pending_tips(deps, info, value)
+        }
+        ExecuteMsg::RemovePendingTip {
+            application,
+            handle,
+        } => remove_pending_tip(deps, info, application, handle),
+    }
+}
+
+pub fn send_tip(
+    deps: DepsMut<DesmosQuery>,
+    env: Env,
+    info: MessageInfo,
+    application: String,
+    handle: String,
+) -> Result<Response<DesmosMsg>, ContractError> {
+    let querier = ProfilesQuerier::new(deps.querier.deref());
+    let sender = info.sender;
+    let funds = sum_coins_sorted(info.funds)?;
+
+    if funds.is_empty() {
+        return Err(ContractError::EmptyTipAmount {});
+    }
+
+    // Query users that have that application linked to their accounts.
+    let response = querier.query_application_link_owners(
+        Some(application.clone()),
+        Some(handle.clone()),
+        None,
+    )?;
+
+    // Fail if there are more then 1 users with the same application link.
+    if response.owners.len() > 1 {
+        return Err(ContractError::ToManyOwners {});
+    }
```

Comment:
Should we remove it? From my side, all addresses linked to the application should be considered the valid to send the tips. What do you think?

Reply:
Ok, I can add a `owner_index: Option<Uint64>` to `SendTip` so that the user can select to witch addres the tip will be sent and if is `None` I will pick the first owner item.